src/ucp/wireup/address.c

```diff
@@ -1464,11 +1464,17 @@ ucs_status_t ucp_address_unpack(ucp_worker_t *worker, const void *buffer,
                   addr_version, addr_flags);
     }
 
-    if (unpack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_UUID) {
-        ucs_assert(addr_flags & UCP_ADDRESS_HEADER_FLAG_WORKER_UUID);
-    }
-
-    if (addr_flags & UCP_ADDRESS_HEADER_FLAG_WORKER_UUID) {
+    if ((unpack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_UUID) ||
+        (addr_flags & UCP_ADDRESS_HEADER_FLAG_WORKER_UUID)) {
```

Comment:
can we do this check according to address version?

Reply:
ok

Reply:
fixed

Reply:
does it make sens to do
``
if ((unpack_flags & UCP_ADDRESS_PACK_FLAG_WORKER_UUID && version == V1) ||
        (addr_flags & UCP_ADDRESS_HEADER_FLAG_WORKER_UUID)) {
```

Reply:
what extra case would it handle?

Reply:
do not unpack UUID if not present in v2

Reply:
but this situation is not supposed to happen and is guarded by the assert above. 
Maybe make that assert to be `always`?

Reply:
i think better ignore than assert, it will make future extensions possible

Reply:
so, remove the assert and add the addr check to if?